tests/test_gradient_accumulation.py

```diff
@@ -0,0 +1,65 @@
+import tempfile
```

Comment:
How hard would it be to add a test that verifies that the gradient for a toy model is the same with two different (but effectively equivalent) batch size/accumulation combinations? Given our current testing standards, I'm okay merging without this, but it'd definitely be good to have.

Reply:
Right now this would be hard. But I agree that this is important. I'll open an issue on it.

Reply:
Open when ready.